### PR TITLE
[Silabs] PSA Crypto PAL customized for EFR32.

### PR DESCRIPTION
**Pull request by rcasallas-silabs**
* Added flag to allow customized Crypto PAL PSA.
* Opaque PSA key now used in EFR32 instead of raw key.

Tested on EFR32MG24.

### CODE REVIEW

File: src/crypto/crypto.gni [hunk 1]
```diff
@@ -22,6 +22,9 @@ declare_args() {
 
   # Use PSA Spake2+ implementation. Only used if chip_crypto == "psa"
   chip_crypto_psa_spake2p = false
+
+  # Use custom PSA crypto.
```
[the remark this answers is not in the document]
Comment (rcasallas-silabs) [reply src/crypto/crypto.gni:26]: I am open to suggestions, but if you select `crypt_crypto == "psa"`, then you include `CHIPCryptoPALPSA.cpp`, which we don't want. If you select `crypt_crypto != "psa"`, then you include `RawKeySessionKeystore.cpp` which we don't want either. If you know a way to include `PSAOperationalKeystore.cpp` but not `CHIPCryptoPALPSA.cpp`, please let me know.

**Review by jmartinez-silabs — APPROVED** [review]

File: src/crypto/BUILD.gn [hunk 2]
```diff
@@ -38,12 +38,24 @@ assert(
         chip_crypto == "platform",
     "Please select a valid crypto implementation: mbedtls, psa, openssl, boringssl, platform")
 
+if (chip_crypto_keystore == "") {
+  if (chip_crypto == "psa") {
+    chip_crypto_keystore = "psa"
+  } else {
+    chip_crypto_keystore = "raw"
+  }
```
Comment (tcarmelveilleux) [inline src/crypto/BUILD.gn:46]: Add "provided_by_app" which includes zero new files and disables the unit tests for the keystore.

File: src/platform/silabs/efr32/CHIPCryptoPALPsaEfr32.cpp [hunk 3]
```diff
@@ -161,121 +155,124 @@ static int timeCompare(mbedtls_x509_time * t1, mbedtls_x509_time * t2)
     return 0;
 }
 
+bool isBufferNonEmpty(const uint8_t * data, size_t data_length)
```
Comment (tcarmelveilleux) [inline src/platform/silabs/efr32/CHIPCryptoPALPsaEfr32.cpp:158]: nit: IsBufferNonEmpty/IsValidTag/TimeCompare to match the usual style.

File: src/platform/silabs/efr32/CHIPCryptoPALPsaEfr32.cpp [hunk 4]
```diff
@@ -161,121 +155,124 @@ static int timeCompare(mbedtls_x509_time * t1, mbedtls_x509_time * t2)
...
-        psa_driver_wrapper_aead_decrypt(&attr, key.As<Symmetric128BitsKeyByteArray>(), sizeof(Symmetric128BitsKeyByteArray),
-                                        PSA_ALG_AEAD_WITH_SHORTENED_TAG(PSA_ALG_CCM, tag_length), Uint8::to_const_uchar(nonce),
-                                        nonce_length, Uint8::to_const_uchar(aad), aad_len, allocated_buffer ? buffer : ciphertext,
-                                        ciphertext_len + tag_length, plaintext, ciphertext_len, &output_length);
-
-    if (allocated_buffer)
+    else
     {
-        memset(buffer, 0, ciphertext_len + tag_length);
+        ChipLogDetail(Crypto, "AES_CCM_decrypt: Using aad == null path");
```
Comment (tcarmelveilleux) [inline src/platform/silabs/efr32/CHIPCryptoPALPsaEfr32.cpp:254]: Is this relevant?

**Review by tcarmelveilleux — APPROVED** [review]

**Review by mkardous-silabs — APPROVED** [review]

**Pull request merged**